### DISCUSSION TITLE
Host blocked from southwest.com?

### DISCUSSION THREAD

**Discussion by derrick-tao** [jdholtz/auto-southwest-check-in#95] [Q&A]
Have any of you been banned or blocked from accessing southwest.com website due to running this script? currently my docker machine and laptop has been blocked presumingly because i've used this script on those machines but i still have access from other machines on my local network. so seems like my WAN ip isn't blocked but there are doing something else to track the machine. any ideas to get around this?

**Comment by jdholtz** [comment]
I have not run into this issue. That is strange that it works for other devices on your local network. Do you get the same result on a VPN (or the Tor browser) on these machines?

**Reply by derrick-tao** [reply]
Weird. I wonder if it's because I was tinkering around with the code awhile back and making multiple calls to the southwest website. When I switch on my vpn, I'm able to access southwest.

**Reply by jdholtz** [reply]
What retrieval interval are you currently using? I would be very surprised if they blocked your IP specifically if you only request to their website a few times every 24 hours (assuming that’s the interval). 

They have millions of users so I’d be curious to see how they are able to pick you specifically. They could also be using your User Agent. Do you get the same issue if you try on Firefox with the same machine?

**Reply by derrick-tao** [reply]
I was using 24 hour interval so nothing crazy. Even when tinkering it should have been only a couple of consecutive calls for no more than an hour. If I use a different browser (Firefox and safari) or even incognito mode, it's the same blocked message. So I'm not exactly sure how they are tracking.

**Reply by jdholtz** [reply]
You could also try to change the IP address on your machine to see if it starts working again.

**Reply by derrick-tao** [reply]
I tried changing my machine's local ip was still blocked. They must be doing a bit more than ip blocking.

**Reply by jdholtz** [reply]
What’s the status code you are receiving?

**Reply by derrick-tao** [reply]
I'm receiving 403 forbidden.  

<img width="528" alt="Screenshot 2023-06-05 at 10 12 21 AM" src="https://github.com/jdholtz/auto-southwest-check-in/assets/288877/e1b2b91f-ec5c-4ea2-993d-8fefbe5e4ca9">

**Reply by derrick-tao** [reply]
Looks like all devices are blocked from my WAN. I'll have to request a new IP from my ISP. =(

**Reply by jdholtz** [reply]
I’m surprised that Southwest would put this much focus on one user (assuming they blocked your IP). Let me know when you get it working again.

**Reply by blakeaa827** [reply]
Been thinking about this a bit.

Is it possible your clock is not sync'ed to an NTP service?

I'm wondering if your clock was a little bit ahead of schedule and the app triggered early?

If for example your computer is a minute fast, the app might have spammed the server for a full minute prior to the check-in being available...

If that were the case, it might make sense to put in some sort of throttle so that if there is no response after say 5-6 attempts, the script throttled attempts back to 1 per 1 or 2 seconds.

That might be enough to keep from getting your IP banned for too many attempts prior to check-in being available.

**Reply by jdholtz** [reply]
This script limits the number of attempts to 20 for a request to Southwest which is a simple protection against this problem. I’ve had multiple times hitting 20 attempts while testing and have not gotten IP banned. However, we don’t know the exact cause of the 403 Forbidden, so it may be possible another issue was going on besides an IP ban.

**Reply by dpbraune** [reply]
Just how dependent is the app to clock time? I can see the issue with a clock that is ahead in time but what if your clock is behind  My laptops sync with NIST.GOV every 24 hours (I believe).

**Reply by jdholtz** [reply]
> Just how dependent is the app to clock time?

It is fully dependent on the system’s clock time. This script works best on servers that run 24/7, so you can just let it run forever. However, you can run it on your laptop just fine as long as you don’t put it to sleep while the script is running (in that case, you will need to restart the script so the sleep time of the script is synchronized, as it doesn’t count down while your laptop is asleep).
